src/Components/Server/src/Circuits/RemoteJSRuntime.cs

```diff
@@ -47,7 +62,76 @@ protected override void BeginInvokeJS(long asyncHandle, string identifier, strin
                     "attempted during prerendering or while the client is disconnected.");
             }
 
+            if (identifier != "DotNet.jsCallDispatcher.endInvokeDotNetFromJS")
+            {
+                // Filter the default implementation of endInvokeDotNetFromJS, we could change this to dispatch to a client method directly
+                // instead of going through JS interop.
```

Comment:
I was going to suggest doing this now to avoid the breaking change on `JSRuntimeBase` later, but I know it's super inconvenient because it's a cross-repo change.

I *guess* that technically we could do this in the future as a non-breaking change by adding a new `virtual` (non-`abstract`) method on `JSRuntimeBase` for `EndInvokeJS` whose base-class implementation makes a JS-interop call to `DotNet.jsCallDispatcher.endInvokeDotNetFromJS`. We could override it in `RemoteJSRuntime` to call some entirely separate JS-side hub client method like `JS.EndInvokeJS` and then wouldn't need the check on L65 here.

So if you feel the extra cost of doing this now is prohibitive, I'm fine with leaving it.

Reply:
We can also simply avoid calling base on `EndInvokeDotNet` and simply call a method on the client. That would have the same effect. I'm ok going there if you are. I just didn't want to waste the effort as is yet another thing add to the PR.

Reply:
Oh, I think my previous comment was slightly confused and should have said `EndInvokeDotNet` when I said `EndInvokeJS`.

If so, that would mean you and I are basically suggesting the same thing. The only difference might be: if we're changing this now, then I'd suggest `EndInvokeDotNet` should become `abstract`, not `virtual`. Each implementation provides its own transport for `EndInvokeDotNet` - there isn't a built-in default. Our WebAssembly one can make a JS interop call like the existing one already does, and our RemoteJSRuntime can call a method on the client.

Reply:
I updated the server-side implementation to use a custom client method. I'll make it abstract and push the implementation down to the webassembly version on the other PR.

Reply:
done